great_ape_safe/ape_api/balancer.py

```diff
@@ -280,6 +280,20 @@ def stake_all(self, pool, destination=None, dusty=False):
         self.stake(pool, mantissa, destination, dusty)
 
 
+    def unstake(self, pool, mantissa, claim=True):
+        gauge = self.safe.contract(self.gauge_factory.getPoolGauge(pool))
+        bal_pool_before = pool.balanceOf(self.safe)
+        gauge.withdraw(mantissa, claim)
+        assert pool.balanceOf(self.safe) >= bal_pool_before
+
+
+    def unstake_all(self, pool, claim=True):
+        gauge = self.safe.contract(self.gauge_factory.getPoolGauge(pool))
+        bal_pool_before = pool.balanceOf(self.safe)
+        gauge.withdraw(gauge.balanceOf(self.safe), claim)
+        assert pool.balanceOf(self.safe) >= bal_pool_before
```

Comment:
If gauges are 1-1 shouldn't this be exactly 
`gauge_bal = gauge.balanceOf(self.safe)`
`pool.balanceOf(self.safe) == bal_pool_before + gauge_bal )`
?